jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -151,7 +151,11 @@ ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 stri
                                 lit_utf8_size_t string_size) /**< string size */
 {
   JERRY_ASSERT (string_p != NULL || string_size == 0);
-  JERRY_ASSERT (lit_is_valid_cesu8_string (string_p, string_size));
+  if (!lit_is_valid_cesu8_string (string_p, string_size))
```

Comment:
The input source (file) is in a valid utf8 format but not in cesu8 format, so `lexer_construct_regexp_object` calls the `ecma_new_ecma_string_from_utf8` with an utf8 string and the assert fails. I have concerns, because this might happen a lot of places where `ecma_new_ecma_string_from_utf8` is called. For example what happens if such strings appears in a variable name or string literals, etc.